### PR TITLE
Minor style and code tweaks and use where syntax for misc methods

### PR DESCRIPTION
**Pull request by musm**

### CODE REVIEW

File: base/linalg/bidiag.jl [hunk 1]
```diff
@@ -493,21 +493,21 @@ function A_mul_B_td!(C::AbstractMatrix, A::AbstractMatrix, B::BiTriSym)
     C
 end
 
-SpecialMatrix = Union{Bidiagonal, SymTridiagonal, Tridiagonal}
+const SpecialMatrix = Union{Bidiagonal,SymTridiagonal,Tridiagonal}
```
Comment (musm) [inline base/linalg/bidiag.jl]: do `Union` need to have `const` or is that implied

Comment (StefanKarpinski) [inline base/linalg/bidiag.jl]: Yes, these need to be const since there's not `{ }` on the left hand side.

File: base/linalg/bidiag.jl [hunk 2]
```diff
@@ -630,14 +630,14 @@ SpecialArrays = Union{Diagonal,
 end
 
 # for historical reasons:
```
Comment (tkelman) [inline base/linalg/bidiag.jl]: unrelated, but we should remember to check how long this has been around and see about deprecating it later

**Comment by musm** [comment]
timed out no i686 appveyor otherwise tests pass

File: base/linalg/bidiag.jl [hunk 3]
```diff
@@ -630,14 +630,14 @@ SpecialArrays = Union{Diagonal,
...
-fill!(D::Diagonal, x) = fillslots!(D, x);
+fill!(a::AbstractTriangular, x) = fillslots!(a, x)
+fill!(D::Diagonal, x) = fillslots!(D, x)
 
 _small_enough(A::Bidiagonal) = size(A, 1) <= 1
 _small_enough(A::Tridiagonal) = size(A, 1) <= 2
 _small_enough(A::SymTridiagonal) = size(A, 1) <= 2
 
-function fill!(A::Union{Bidiagonal, Tridiagonal, SymTridiagonal} ,x)
+function fill!(A::Union{Bidiagonal, Tridiagonal, SymTridiagonal}, x)
```
Comment (Sacha0) [inline base/linalg/bidiag.jl]: Extra spaces in the union?

**Pull request merged**